ydb/docs/en/core/reference/ydb-sdk/topic.md

```diff
@@ -776,6 +783,139 @@ All the metadata provided when writing a message is returned to a consumer with
 
   ```
 
+- Java
+
+  To take advantage of message metadata feature, build messages with builder. You can add `MetadataItem` to a message or set a `List` of `MetadataItem`s. Each item consists of a key of type `String` and a value of type `byte[]`:
+
+  ```java
+  writer.send(
+          Message.newBuilder()
+                  .addMetadataItem(new MetadataItem("meta-key", "meta-value".getBytes()))
+                  .addMetadataItem(new MetadataItem("another-key", "value".getBytes()))
+                  .build()
+  ```
+
+  While reading, metadata can be received from a `Message` with `getMetadataItems()` method:
+
+  ```java
+  Message message = reader.receive();
+  List<MetadataItem> metadata = message.getMetadataItems();
+  ```
+
+{% endlist %}
+
+### Write in transaction {#write-tx}
+
+{% list tabs %}
+
+- Java (sync)
+
+  [Example on GitHub](https://github.com/ydb-platform/ydb-java-examples/blob/develop/ydb-cookbook/src/main/java/tech/ydb/examples/topic/transactions/TransactionWriteSync.java)
+
+  Transaction can be set in `SendSettings` of `send` method while sending a message.
+  Such message will be written on transaction commit.
+
+  ```java
+  // creating session in table service
+  Result<Session> sessionResult = tableClient.createSession(Duration.ofSeconds(10)).join();
+  if (!sessionResult.isSuccess()) {
+      logger.error("Couldn't get session from pool: {}", sessionResult);
+      return; // retry or shutdown
+  }
+  Session session = sessionResult.getValue();
+  // creating transaction in table service
+  // this transaction is not yet active and has no id
+  TableTransaction transaction = session.createNewTransaction(TxMode.SERIALIZABLE_RW);
```

Comment:
Probably, other transaction modes aren't supported here? It is worth mentioning if that's the case.

Reply:
All existing tx modes are supported. The most common one was chosen just to showcase how transactions and topics interact

Reply:
@pnv1 so writing to topics is allowed even for read-only transactions? if that's the case, it requires even more explanation on how this works

Reply:
I've been told that only serializable RW transactions are supported by topic service on server size atm. So your assumption was correct. I'll add this information to this documentation